_posts/2019-01-23-multifile-python-functions.md

```diff
@@ -0,0 +1,213 @@
+---
+title: Multifile python functions
+description: "Tidy python in an OpenFaaS world; or, what happens when you want to split your Python function into multiple modules?"
+date: 2019-01-23
+image: /images/multifile-python-function/pexels-book-stack-books-bookshop-264635.jpg
+categories:
+  - python
+author_staff_member: lucas
+dark_background: true
+---
+
+As a core contributor to OpenFaaS, I hangout in the OpenFaaS Slack a lot, you can [join us here][openfaas-slack-singnup]. Most recently that meant trying to answer this question
+
+> [How can I] write a function in python that uses multiple source files so that the main `handler.py` is able to import helper functions from separate source files.
```

Comment:
` [How can I]` ->  `How can I`

No need to quote verbatim.

Reply:
You may add a commentary line here saying _why_ a user would want to split out into different files - i.e. for testing, scaling, reuse. You may want to link to the Python docs.

Say what version of Python you are working with. i.e. 3

Reply:
To your first comment, "How can i" is a modification of a quote, the rest if of the statement is a true quote from slack.

To you rsecond comment, I explicitly mention python 3 below.  Given that Python 2 is end of life this year, I don't even want to address it.  We should always pick Python 3 as our default Python.

I will add a comment on motivation

Reply:
If it's a quote then it should be in double quotes and use the user's name to attribute it.

We support both Python 2.7 and Python 3. Please mention that this example will be for a specific version of template up top.

Reply:
where up top should it go? It is already the first thing mentioned in the next section.  I can add a small section about how relative imports are done in python 2.  The code is the same but to really ensure it is safe you should put `from __future__ import absolute_import` at the top of `wordcount.py` and `index.py`

Reply:
I have added an explicit mention of python3 in the intro and a section about python 2 at the end